presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ApplyConnectorOptimization.java

```diff
@@ -292,4 +301,26 @@ private static <T> boolean containsAll(Set<T> container, Collection<T> test)
         }
         return true;
     }
+
+    private static class ConnectorToInternalJoinRewriter
+            extends SimplePlanRewriter<Void>
+    {
+        @Override
+        public PlanNode visitConnectorJoinNode(ConnectorJoinNode node, RewriteContext<Void> context)
+        {
+            return new JoinNode(node.getSourceLocation(),
+                    node.getId(),
+                    Optional.empty(),
+                    node.getType(),
+                    context.rewrite(node.getSources().get(0)),
+                    context.rewrite(node.getSources().get(1)),
+                    ImmutableList.copyOf(node.getCriteria()),
+                    node.getOutputVariables(),
+                    node.getFilters().stream().reduce((a, b) -> new SpecialFormExpression(SpecialFormExpression.Form.AND, BooleanType.BOOLEAN, a, b)),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
```

Comment:
The `ConnectorJoinNode` currently doesn't allow setting it since I just relied on automatic. I think it can be added when a connector wants to set it?

Reply:
But this will remove the decision from DetermineJoinDistributionType optimizer?

Reply:
The `ConnectorJoinNode` is converted as soon as it's returned from the connector in `ApplyConnectorOptimization` (logical). Which is before `DetermineJoinDistributionType`.

Reply:
How about ApplyConnectorOptimization (physical)? 

> The ConnectorJoinNode currently doesn't allow setting it since I just relied on automatic

Why it does not allow? Can you elaborate?

Reply:
As I understand, ApplyConnectorOptimization(physical) did not contains optimizer `IcebergEqualityDeleteAsJoin`, so it would never meet and rewrite `ConnectorJoinNode`.

Reply:
Exactly. 
If you want me to allow it I can move `DistributionType` to the SPI package as well. I just didn't need it so I didn't want to move more classes than needed. But let me know if you prefer I move it as well.

Reply:
Sounds like too much assumption and dependencies behind this.

> move DistributionType to the SPI package as well

I think it will be great, if there are no other reasons which prevents it.